runner/debug/vault.go

```diff
@@ -0,0 +1,123 @@
+package debug
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/hcdiag/op"
+	"github.com/hashicorp/hcdiag/redact"
+	"github.com/hashicorp/hcdiag/runner"
+)
+
+var _ runner.Runner = VaultDebug{}
+
+// VaultDebugConfig is a config struct for VaultDebug runners
+type VaultDebugConfig struct {
+	Compress        string
+	Duration        string
+	Interval        string
+	LogFormat       string
+	MetricsInterval string
+	Targets         []string
+	Redactions      []*redact.Redact
+}
+
+// VaultDebug represents a VaultDebug runner
+type VaultDebug struct {
+	Compress        string           `json:"compress"`
+	Duration        string           `json:"duration"`
+	Interval        string           `json:"interval"`
+	LogFormat       string           `json:"logformat"`
+	MetricsInterval string           `json:"metricsinterval"`
+	Targets         []string         `json:"targets"`
+	Redactions      []*redact.Redact `json:"redactions"`
+
+	output string
+}
+
+func (d VaultDebug) ID() string {
+	return "VaultDebug"
+}
+
+func NewVaultDebug(cfg VaultDebugConfig, tmpDir string, debugDuration time.Duration, debugInterval time.Duration) *VaultDebug {
+	dbg := VaultDebug{
+		// No compression because the hcdiag bundle will get compressed anyway
```

Comment:
I think the product debugs get compressed today, don't they? I would probably rather either keep it consistent or else be sure to capture this change in the changelog so that users have a way of expecting the change in behavior without reading the code.

Reply:
Yep, I originally planned to have the SimpleDebug runner be a drop-in replacement, with the more configurable runners mirroring product debug defaults. That's getting removed, and I'll change the default to `-compress=true`.